data_steward/retraction/participant_row_counts.py

```diff
@@ -0,0 +1,249 @@
+    :param dataset_id: identifies the dataset
+    :param pid_source: identifies the source of pids
+    :param table_df: dataframe from BQ INFORMATION_SCHEMA.COLUMNS
+    :param for_rdr: indicates if query is for an RDR dataset. If True, set ehr_counts = 0,
+        else set ehr_counts to "COUNT(*)" for unioned_ehr and other datasets that need investigation
+    :return: query:
+    """
+    query_list = []
+    pid_sql_expr = ru.get_pid_sql_expr(pid_source)
+    tables_with_pid = ru.get_pid_tables(table_df)
+
+    # Unioned EHR or generic dataset
+    for table in tables_with_pid:
+        ehr_count = 0 if for_rdr else "COUNT(*)"
+        tmpl = Template(consts.PID_TABLE_COUNT).render(project=project_id,
+                                                       dataset=dataset_id,
+                                                       table=table,
+                                                       pids_expr=pid_sql_expr,
+                                                       ehr_count=ehr_count)
+        query_list.append(tmpl)
+    query = consts.UNION_ALL.join(query_list)
+    return query
+
+
+def get_ehr_query(project_id, dataset_id, pid_source, hpo_id, table_df):
+    """
+    Get query to determine all row counts and ehr row counts from ehr datasets,
+    returns unioned_ehr query if hpo_id is set to 'none', since ehr datasets also contain unioned tables
+
+    :param project_id: identifies the project
+    :param dataset_id: identifies the dataset
+    :param pid_source: identifies the source of pids
+    :param hpo_id: identifies the hpo site that submitted the pids, can be set to 'none' for unioned tables query
+    :param table_df: dataframe from BQ INFORMATION_SCHEMA.COLUMNS
+    :return: query:
+    """
+    query_list = []
+    pid_sql_expr = ru.get_pid_sql_expr(pid_source)
+    tables_with_pid = ru.get_pid_tables(table_df)
+
+    # EHR
+    unioned_tables = [
+        table for table in tables_with_pid if common.UNIONED_EHR in table
+    ]
+
+    tables_to_consider = unioned_tables
+
+    if hpo_id != 'none':
+        hpo_tables_with_pid = [
+            table for table in tables_with_pid if hpo_id in table
+        ]
+        tables_to_consider += hpo_tables_with_pid
+
+    for table in tables_to_consider:
+        ehr_count = "COUNT(*)"
+        tmpl = Template(consts.PID_TABLE_COUNT).render(project=project_id,
+                                                       dataset=dataset_id,
+                                                       table=table,
+                                                       pids_expr=pid_sql_expr,
+                                                       ehr_count=ehr_count)
+        query_list.append(tmpl)
+    query = consts.UNION_ALL.join(query_list)
+    return query
+
+
+def count_pid_rows_in_dataset(project_id, dataset_id, hpo_id, pid_source):
+    """
+    Returns df containing tables which have non-zero counts of participant rows for pids in pids_source
```

Comment:
There are a lot of tables in our datasets, making it difficult to sort through them visually if everything was logged. I thought it would be better to highlight the 'tables of interest' in a sense, by skipping those with zero counts. I could add a note saying so.

Reply:
Ah I see. How about sorting them by count DESC? Zeroes are still valuable information. They indicate that no records should be deleted from a table after retraction. We would be suspicious and react if we found that records did in fact drop in those tables. Zeroes could also clue us in to something wrong with the pid list.

Reply:
Yeah I think it is valuable information as well. I'll add this.